asmcli/asmcli.sh

```diff
@@ -1025,12 +1048,39 @@ validate_args() {
     fi
     readonly "${REQUIRED_ARG}"
   done <<EOF
+MODE
+EOF
+
+  local CLUSTER_DETAIL_SUPPLIED=0
+  local CLUSTER_DETAIL_VALID=1
+  while read -r REQUIRED_ARG; do
+    if [[ -z "${!REQUIRED_ARG}" ]]; then
+      CLUSTER_DETAIL_VALID=0
+    else
+      CLUSTER_DETAIL_SUPPLIED=1
+    fi
+  done <<EOF
 CLUSTER_LOCATION
 CLUSTER_NAME
 PROJECT_ID
-MODE
 EOF
 
+  if [[ "${CLUSTER_DETAIL_SUPPLIED}" -eq 1 && "${CLUSTER_DETAIL_VALID}" -eq 0 ]]; then
+    fatal_with_usage "Missing one or more required options for [CLUSTER_LOCATION|CLUSTER_NAME|PROJECT_ID]"
```

Comment:
Pardon me if I am mistaken here, but to me we shouldn't fatal here. The point of having `MISSING_ARGS` in my opinion is to give out as many warnings as possible before exiting to save the users' time. Do you think it's reasonable to `warn` here and let `MISSING_ARGS` handle the `fatal_with_usage` case?

Reply:
thanks for pointing that out, rearranged and changed to warn.